coordinator/src/cli.rs

```diff
@@ -34,4 +40,40 @@ impl Opts {
 
         Ok(data_dir)
     }
+
+    /// Returns a list of addresses under which the node can be reached. Note this is used for the
+    /// node announcements.
+    pub fn p2p_announcement_addresses(&self) -> Vec<NetAddress> {
+        let mut addresses: Vec<NetAddress> = vec![];
+        if !self.p2p_address.ip().is_unspecified() {
+            addresses.push(build_net_address(
+                self.p2p_address.ip(),
+                self.p2p_address.port(),
+            ));
+        } else {
+            // Announcing the node on an unspecified ip address does not make any sense.
+            tracing::warn!("Skipping node announcement on '0.0.0.0'.");
+        }
+
+        if !self.skip_local_network_announcement {
+            let local_ip = local_ip().expect("to get local ip address");
+            tracing::info!("Adding node announcement within local network {local_ip}. Do not use for production!");
+            addresses.push(build_net_address(local_ip, self.p2p_address.port()));
```

Comment:
❓ Won't you end up pushing the same `LOCAL_IP:port` address _twice_ if `skip_local_network_announcement == false` and the `p2p_address` is the default one?

Reply:
No sure if I follow, the default address of p2p_address is `0.0.0.0`. I guess if you set the p2p_address to your local_ip and do not `skip_local_network_announcement` you will end up pushing the same address twice.

Do suggest to add a handling for that? Looks to me more like a miss configuration.

Reply:
I misread what we were doing above and forgot that it's pointless to announce on `0.0.0.0`. I get it now! Sorry for the confusion.